src/service/url-replacements-impl.js

```diff
@@ -49,6 +49,28 @@ let AsyncResolverDef;
 let ReplacementDef;
 
 /**
+ * Returns a encoded URI Component, or an empty string if the value is nullish.
+ * @param {ResolverReturnDef} val
+ * @return {string}
+ */
+function encodeValue(val) {
+  if (val == null) {
+    return '';
+  }
+  return encodeURIComponent(val);
+};
+
+/**
+ * Detects the protocol of the url, if there is one.
+ * @param {string} url
+ * @return {string}
+ */
+function protocolOfUrl(url) {
+  const match = /^.*:/.exec(url);
```

Comment:
I specifically didn't want to do this, since these replacements can happen many times (analytics). Each new URL (`RANDOM` is a substitution) would be cached.